src/AssetsPage/AssetsTable/AssetsTable.test.jsx

```diff
@@ -24,31 +28,354 @@ const defaultProps = {
       display_name: 'bird.json',
       id: 'bird.json',
     },
+    {
+      display_name: 'fish.doc',
+      id: 'fish.doc',
+    },
   ],
   assetsParameters: {
-    assetTypes: {
-      courseId: 'test-course',
-      page: '0',
-      pageSize: '50',
-      sort: 'sort',
-    },
+    courseId: 'test-course',
+    page: '0',
+    pageSize: '50',
   },
   assetsStatus: {},
-  deleteAsset: () => {},
   clearAssetsStatus: () => {},
+  deleteAsset: () => {},
+  updateSort: () => {},
 };
 
-let wrapper;
-
 describe('<AssetsTable />', () => {
-  it('renders correct number of assets', () => {
-    wrapper = mount(
-      <AssetsTable
-        {...defaultProps}
-      />,
-    );
-    // + 1 because it includes row item for headings
-    expect(wrapper.find('tr')).toHaveLength(defaultProps.assetsList.length + 1);
+  describe('renders', () => {
+    it(' correct number of assets', () => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+      // + 1 because it includes row item for headings
+      expect(wrapper.find('tr')).toHaveLength(defaultProps.assetsList.length + 1);
+    });
+    it('correct number of trashcan buttons', () => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+
+      expect(wrapper.find('button').filterWhere(button => button.hasClass('fa-trash'))).toHaveLength(defaultProps.assetsList.length);
+    });
+  });
+  describe('onSortClick', () => {
+    let updateSortSpy;
+
+    beforeEach(() => {
+      updateSortSpy = jest.fn();
+    });
+
+    const sortableAssetProps = [
+      {
+        sort: 'date_added',
+        direction: 'desc',
+      },
+      {
+        sort: 'date_added',
+        direction: 'asc',
+      },
+      {
+        sort: 'date_added',
+        direction: 'desc',
+      },
+    ];
+
+    const assertValues = [
```

Comment:
Basically, these are values needed by each test.

Reply:
I'm a little concerned about having 2 dictionaries that are implicitly linked. Could we iterate over a single, more complex object to avoid using `index`? Something like:

```
const testData = [
  {
    sortProps: {
      sort: 'date_added',
      direction: 'desc'
    },
    expectedValues: {
      <appropriate key: value pairs>
    }
  },
  ...
]
```

Then the test loop becomes `testData.forEach`. Just an idea, if this works currently and it's difficult to switch don't sweat it.